server/auth/registrar.go

```diff
@@ -25,7 +30,372 @@ var (
+//     the required number of confirmations, and send 'postbond' again.
+//   - If the located transaction has the required number of confirmations,
+//     the bond is stored in the DB as active, and a response is sent.
+//   - Otherwise, a response is sent with the observed confirmation count, and a
+//     coin waiter is started to watch for the txn to reach the required number
+//     of confirmations. The bond is stored in DB as pending. When the
+//     transaction reaches the required confirmations, it is flagged active in
+//     the DB and a 'bondconfirmed' notification is sent.
+//
+// NOTE: This presently only supports Decred bonds.
+func (auth *AuthManager) handlePostBond(conn comms.Link, msg *msgjson.Message) *msgjson.Error {
+	postBond := new(msgjson.PostBond)
+	err := msg.Unmarshal(&postBond)
+	if err != nil || postBond == nil {
+		return msgjson.NewError(msgjson.BondError, "error parsing postbond request")
+	}
+
+	// TODO: allow different assets for bond, switching parse functions, etc.
+	assetID := postBond.AssetID
+	if assetID != 42 {
+		return msgjson.NewError(msgjson.BondError, "only DCR bonds supported presently")
+	}
+
+	// Create an account.Account from the provided pubkey.
+	acct, err := account.NewAccountFromPubKey(postBond.AcctPubKey)
+	if err != nil {
+		return msgjson.NewError(msgjson.BondError, "error parsing account pubkey: "+err.Error())
+	}
+	acctID := acct.ID
+
+	// Authenticate the message for the supposed account.
+	sigMsg := postBond.Serialize()
+	err = checkSigS256(sigMsg, postBond.SigBytes(), acct.PubKey)
+	if err != nil {
+		return &msgjson.Error{
+			Code:    msgjson.SignatureError,
+			Message: "signature error: " + err.Error(),
+		}
+	}
+
+	// A bond's lockTime must be after bonExpiry from now.
+	lockTimeThresh := time.Now().Add(auth.bondExpiry)
+
+	// Decode raw tx, check fee output (0) and account commitment output (1).
+	bondCoinID, amt, bondAddr, bondPK, lockTime, commitAcct, err :=
+		auth.parseBondTx(postBond.BondVer, postBond.BondTx, postBond.BondScript) // i.e. dcr.ParseBondTx presently, but should switch on asset
+	if err != nil {
+		return msgjson.NewError(msgjson.BondError, "invalid bond transaction: %v", err)
+	}
+	if amt < int64(auth.minBond) {
+		return msgjson.NewError(msgjson.BondError, "insufficient bond amount %d, needed %d", amt, auth.minBond)
+	}
+	if lockTime < lockTimeThresh.Unix() {
+		return msgjson.NewError(msgjson.BondError, "insufficient lock time %d, needed at least %d", lockTime, lockTimeThresh.Unix())
+	}
+
+	// Must be equal to account ID computed from pubkey in the PayFee message.
+	if commitAcct != acctID {
+		return msgjson.NewError(msgjson.BondError, "invalid bond transaction - account commitment does not match pubkey")
+	}
+
+	// Verify that the user has the private keys to the bond pubkey by verifying
+	// the BondSig against the message (acct ID) and bond script's pubkey hash.
+	msgHash := sha256.Sum256(acctID[:])
```

Comment:
This is how the user shows that the bond *output* is actually theirs.  The `commitAcct` comes from the second output with the OP_RETURN, but the first output (or the corresponding redeem script) does not directly reference the account to which it should apply.

Reply:
When there is mesh, the user will need to supply this extra signature for every dex? Also, this is always the same, so a server could impersonate a user after seeing this once.

I kind of don't think it is necessary to check or require this. Does it really matter who made the bond? Someone is losing out on opportunity costs for this unique ID.

Reply:
> When there is mesh, the user will need to supply this extra signature for every dex? Also, this is always the same, so a server could impersonate a user after seeing this once.

That's true.  It could be a signature of something else though.

> I kind of don't think it is necessary to check or require this. Does it really matter who made the bond? Someone is losing out on opportunity costs for this unique ID.

OK, I think you guys convinced me this does little. I suppose I'll remove `BondSig`.